src/pages/paymentServiceProviders/CreatePaymentServiceProvider.tsx

```diff
@@ -0,0 +1,207 @@
+import {loginRequest} from "../../authConfig";
+import {PaymentServiceProviderDetails} from "../../../generated/api/PaymentServiceProviderDetails";
+
+interface IProps {
+    match: {
+        params: Record<string, unknown>;
+    };
+    history: {
+        push(url: string): void;
+    };
+}
+
+interface IState {
+    paymentServiceProvider: PaymentServiceProviderDetails;
+    showModal: boolean;
+}
+
+export default class CreatePaymentServiceProvider extends React.Component<IProps, IState> {
+    static contextType = MsalContext;
+
+    service = "/payment-service-providers";
+
+    constructor(props: IProps) {
+        super(props);
+
+        this.state = {
+            paymentServiceProvider: {
+                "abi": "",
+                "agid_psp": false,
+                "bic": "",
+                "enabled": false,
+                "stamp": false,
+                "tax_code": "",
+                "transfer": false,
+                "vat_number": ""
+            } as unknown as PaymentServiceProviderDetails,
+            showModal: false
+        };
+
+        this.discard = this.discard.bind(this);
+        this.save = this.save.bind(this);
+        this.handleChange = this.handleChange.bind(this);
+        this.hideModal = this.hideModal.bind(this);
+    }
+
+    handleChange(event: any) {
+        // eslint-disable-next-line functional/no-let
+        let paymentServiceProvider: PaymentServiceProviderDetails = this.state.paymentServiceProvider;
+        const key = event.target.name as string;
+        const value = event.target.type === "checkbox" ? event.target.checked : event.target.value;
+        paymentServiceProvider = {...paymentServiceProvider, [key]: value};
+        this.setState({paymentServiceProvider});
+    }
+
+    discard(): void {
+        this.setState({showModal: true});
+    }
+
+    hideModal(status: string): void {
+        if (status === "ok") {
+            this.props.history.push(this.service);
+        }
+        this.setState({showModal: false});
+    }
+
+    goBack(): void {
+        this.props.history.push(this.service);
+    }
+
+    save(): void {
+        this.context.instance.acquireTokenSilent({
+            ...loginRequest,
+            account: this.context.accounts[0]
+        })
+            .then((response: any) => {
+                apiClient.createPaymentServiceProvider({
+                    Authorization: `Bearer ${response.accessToken}`,
+                    ApiKey: "",
+                    body: this.state.paymentServiceProvider
+                }).then((response: any) => {
+                    // eslint-disable-next-line no-prototype-builtins
+                    if (response.hasOwnProperty("right")) {
+                        if (response.right.status === 201) {
+                            // eslint-disable-next-line no-console
+                            toast.info("Creazione avvenuta con successo.");
+                            setTimeout(this.goBack.bind(this), 2000);
```

Comment:
why we need a timeout ?

Reply:
This is a smart workaround to redirect user to PSP list section. 
Perhaps we should improve usability in a next step :)